CMakeLists.txt

```diff
@@ -626,7 +626,19 @@ endif()
 ############################
 # TEST
 ############################
-function(build_tests)  
+
+function(add_test_with_working_dir test_name)
```

Comment:
As is the function name does not do what it says

I though at something like:

```
function(add_test_with_working_dir test_name test_dir test_cli)
  add_test(NAME ${test_name} WORKING_DIRECTORY ${test_dir} COMMAND {test_cli})
```

Reply:
There is a reason why CMake recommends to use the function syntax with options instead of fixed arguments. See, e.g., at the bottom of the following page:
https://cmake.org/cmake/help/latest/command/add_test.html
> The command syntax above is recommended over the older, less flexible form

Imho, the changes you are proposing would be a pessimization to the currently proposed implementation (mainly because it would be less flexible).

Please, let me know if this is a show-stopper for you. Otherwise, I'd prefer to not do the change that you are proposing.

As far as the function name is concerned: The new function should be used if tests are added that manually change the working directory to somewhere that doesn't contain the libarpack library. So, the current name isn't misleading imho.
Please, let me know if you prefer a different name for that function.

Reply:
Sorry to bother.
Any feedback on the last comment?

Reply:
> As is the function name does not do what it says

Can you pick a function name what do what it says? `add_test_with_custom_rpath` or similar and explain/comment what you do there "on windows, add path to lib in PATH as rpath is not supported"